### PR TITLE
Accurate donut box sprites

### PR DESCRIPTION
**Pull request by Mothblocks**
## About The Pull Request
The sprite of the donut box will now be dependent on what donuts are inside and in what order.

dreamseeker_2020-07-02T00-13-52
dreamseeker_2020-07-02T00-16-25

## Why It's Good For The Game
Current donut boxes misinform hungry spacemen, and colorful donut boxes are cool.

## Changelog
:cl:
imageadd: Donut boxes will now accurately show what donuts are inside.
/:cl:

### CODE REVIEW

**Comment by trexter555** [comment]
based pr

File: code/game/objects/items/storage/fancy.dm [hunk 1]
```diff
@@ -16,9 +16,6 @@
 
 /obj/item/storage/fancy
 	icon = 'icons/obj/food/containers.dmi'
-	icon_state = "donutbox6"
-	name = "donut box"
-	desc = "Mmm. Donuts."
```
Comment (Jared-Fogle) [inline code/game/objects/items/storage/fancy.dm:21]: Nothing relied on these, I checked.

**Comment by SuperNovaa41** [comment]
Pog!

File: code/game/objects/items/storage/fancy.dm [hunk 2]
```diff
@@ -58,23 +55,55 @@
...
+/obj/item/storage/fancy/donut_box/update_icon_state()
+	if(fancy_open)
+		icon_state = "donutbox_inner"
+	else
+		icon_state = "donutbox"
+
+/obj/item/storage/fancy/donut_box/update_overlays()
+	. = ..()
+
+	cut_overlays()
```
Comment (AnturK) [inline code/game/objects/items/storage/fancy.dm:90]: update_overlays doesn't need to call cut/add it should return new list of overlays.
So you just need do
`. += image()` your donuts here.

**Review by AnturK — CHANGES_REQUESTED** [review]

**Review by AnturK — DISMISSED** [review]

**Comment by mrhugo13** [comment]
The thing that /TG/ needed all along.

**Comment by Doosteroni** [comment]
We solved world hunger!

**Comment by zombiegums** [comment]
Very good PR. Was just thinking about this last night.

**Comment by Rhials** [comment]
I have waited so long for a PR like this, bless you.

**Comment by tmtmtl30** [comment]
we have entered the fucking FUTURE

**Comment by Krysonism** [comment]
Great PR! Although I think the donut icons themselves could be a bit better, i will take a stab at it and PR it to your branch if you agree.

**Comment by Jared-Fogle** [comment]
@Krysonism Go for it.

**Comment by Krysonism** [comment]
> 
> 
> @Krysonism Go for it.

image

This is what i have so far.
I probably have to increase contrast the contrast a bit(in particular i need to make the center darker).

**Comment by MrDoomBringer** [comment]
https://www.youtube.com/watch?v=fxfR5oJoZTc
thread theme

**Pull request merged**